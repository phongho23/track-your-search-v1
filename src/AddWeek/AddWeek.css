.AddWeek {
    height: 200px;
    position: relative;
    color: white;
    margin: auto;
    align-content: center;
    align-items: center;
    text-align: center;
    background-color: #252525;
    width: 70%;
    border-radius: 10px;
  }
  
.AddWeek h2 {
    text-align: center;
  }

.AddWeekItems {
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    justify-content: center;
  }
  