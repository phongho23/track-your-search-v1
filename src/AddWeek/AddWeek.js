import React, { Component } from 'react'
import AppForm from '../AppForm/AppForm'
import ApiContext from '../ApiContext'
import config from '../config'
import './AddWeek.css'

export default class AddWeek extends Component {
  static defaultProps = {
    history: {
      push: () => { }
    },
  }
  static contextType = ApiContext;

  handleSubmit = e => {
    e.preventDefault()
    const week = {
      name: e.target['week-name'].value
    }
    fetch(`${config.API_ENDPOINT}/weeks`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(week),
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
        return res.json()
      })
      .then(week => {
        this.context.addWeek(week)
        this.props.history.push(`/home/week/${week.id}`)
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    return (
      <section className='AddWeek'>
        <h2>Add a Week</h2>
        <AppForm onSubmit={this.handleSubmit}> 
        <div className='AddWeekItems'>
          <div className='field'>
            <label htmlFor='week-name-input'>
              Which week of your job search is this?  
            </label>
            <br />
            <input type='text' id='week-name-input' name='week-name' />
          </div>
          <div className='buttons'>
            <button type='submit'>
              Add Week
            </button>
          </div>
        </div>
        </AppForm>
      </section>
    )
  }
}
