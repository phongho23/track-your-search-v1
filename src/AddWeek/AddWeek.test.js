import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddWeek from './AddWeek'

describe(`AddWeek component`, () => {
  it('renders the complete form', () => {
    const wrapper = shallow(<AddWeek />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
