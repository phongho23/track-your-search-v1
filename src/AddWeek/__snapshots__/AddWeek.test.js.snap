// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddWeek component renders the complete form 1`] = `
<section
  className="AddWeek"
>
  <h2>
    Add a Week
  </h2>
  <AppForm
    onSubmit={[Function]}
  >
    <div
      className="AddWeekItems"
    >
      <div
        className="field"
      >
        <label
          htmlFor="week-name-input"
        >
          Which week of your job search is this?
        </label>
        <br />
        <input
          id="week-name-input"
          name="week-name"
          type="text"
        />
      </div>
      <div
        className="buttons"
      >
        <button
          type="submit"
        >
          Add Week
        </button>
      </div>
    </div>
  </AppForm>
</section>
`;
